File: unit11-hw/ruby/Goose.rb
# Goose.rb | Defines the Goose class which inherits attributes and methods
#   from the Bird superclass.
#
# Begun by: Dr. Jump for CS 315 at King's College.
# Completed by: Robert Lowman
# Date: 11/29/16
####################################################

require_relative 'FlyingBird.rb'

class Goose < FlyingBird

  def call
    'Honk!'
  end

end

File: unit11-hw/python/Goose.py
# Goose.rb | Defines the Goose class which inherits attributes and methods
#   from the Bird superclass.
#
# Completed by: Robert Lowman
# Date: 12/6/16
####################################################
import FlyingBird

class Goose(FlyingBird.FlyingBird):

    def __init__(self, name):
        FlyingBird.__init__(name)
        self.name = name
        
    def call(self):
        return 'Honk!'

    def className(self):
        return 'Goose'
